### PR TITLE
ENH: Add example of OrientImageFilter

### PR DESCRIPTION
**Pull request by blowekamp**
## PR Checklist
- [ ] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [ ] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [ ] Added test (or behavior not changed)
- [ ] Updated API documentation (or API not changed)
- [ ] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [ ] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [ ] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

**Comment by blowekamp** [comment]
Sample output:
```
$ bin/OrientImageFilter ExternalData/Testing/Data/Input/HeadMRVolume.mha test.mha RAS
Input image direction: 
  Matrix (3x3)
    1 0 0 
    0 1 0 
    0 0 1 
The input image has the following orientation: 
	right-to-left anterior-to-posterior inferior-to-superior
The re-orienting to approximately the desired orientation: 
	left-to-right posterior-to-anterior inferior-to-superior
```

Do the have any existing testing data that is 3D with an interesting matrix?

File: Examples/Filtering/OrientImageFilter.cxx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+  // Create and set up an OrientImageFilter
+  using OrientImageFilterType = itk::OrientImageFilter<ImageType, ImageType>;
+  OrientImageFilterType::Pointer orienter = OrientImageFilterType::New();
+  orienter->SetInput(reader->GetOutput());
+  orienter->UseImageDirectionOn();
+  orienter->SetDesiredCoordinateOrientation(desiredCoordinateOrientation);
+
+
+  // Create and set up a writer
```
Comment (dzenanz) [inline Examples/Filtering/OrientImageFilter.cxx:100]: We could simplify this by using `itk::WriteImage(orienter->GetOutput());`

**Review by dzenanz — APPROVED** [review]
Good even as-is.

File: Examples/Filtering/OrientImageFilter.cxx [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+            << std::endl;
+
+  std::cout << "The re-orienting to approximately the desired orientation: "
+            << std::endl;
+  std::cout << '\t' << desiredCoordinateOrientation << std::endl;
+
+
+  // Create and set up an OrientImageFilter
+  using OrientImageFilterType = itk::OrientImageFilter<ImageType, ImageType>;
+  OrientImageFilterType::Pointer orienter = OrientImageFilterType::New();
```
Comment (N-Dekker) [inline Examples/Filtering/OrientImageFilter.cxx:94]: - Minor suggestion: you may replace `OrientImageFilterType::Pointer` with `auto` here! No need to wait for a follow-up to https://github.com/InsightSoftwareConsortium/ITK/pull/4964#issuecomment-2488812821  😃

**Comment by thewtex** [comment]
Nice!

**Pull request merged**